pennylane_sf/gbs.py

```diff
@@ -259,6 +259,16 @@ def jacobian(self, operations, observables, variable_deps):
         """
         self.reset()
 
+        if len(args) == 1:
```

Comment:
Maybe worth using `qml.tape_mode_active()` here?

Reply:
I agree it might seem a bit loose to only check for the number of args, but it was the best way I could think of solving this while still keeping compatibility.

Reply:
How about raising an error if
1) `len(args) == 1` and `args[0]` is not a tape and
2) `len(args) != 1` or `len(args) != 3`?

Reply:
Sure, I could add something like that. Makes sense! :+1: